app/controllers/users_controller.rb

```diff
@@ -0,0 +1,80 @@
+class UsersController < ApplicationController
+  
+  before_action :find_user, only: [:index, :show]
+  
+  def new
+    @user = User.new
+  end
+  
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to root_path
+    else
+      flash[:message] = "Username can't be empty!"
+      render new_user_path
+    end
+  end
+  
+  def index
+    @users = User.all
+  end
+  
+  def show
+    @user = User.find_by(id:params[:id])
+    @votes = Vote.where(user_id: @user.id)
+    head :not_found if @user.nil?
+    
+  end
+  
+  def login_form
+    @user = User.new
+  end
+  
+  def login
+    username = params[:user][:username]
+    @user = User.find_by(username: username)
+    
+    if @user 
+      session[:user_id] = @user.id
+      flash[:message] = "Successfully logged in as #{@user.username}"
+      redirect_to root_path
+    else
+      @user = User.new(username: params[:user][:username])
+      session[:user_id] = @user.id
```

Comment:
Logging in as a new user doesn't work on the first log in because you set `session[:user_id] = @user.id`, but `@user` hasn't been saved in the DB so it doesn't have an ID